### PR TITLE
[develop] Trello-Wj6NIsYU. Авторизация по API

### PR DESCRIPTION
**Pull request by wildeastengineer**
[Авторизация по API](https://trello.com/c/Wj6NIsYU)

- [x] Авторизация
- [x] Регистрация

---

Работа авторизации описана [здесь](https://github.com/wildeastengineer/MKeepAPI/wiki/%D0%A0%D0%B5%D0%B3%D0%B8%D1%81%D1%82%D1%80%D0%B0%D1%86%D0%B8%D1%8F-%7C-%D0%90%D0%B2%D1%82%D0%BE%D1%80%D0%B8%D0%B7%D0%B0%D1%86%D0%B8%D1%8F)

### CODE REVIEW

File: controllers/user.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        UserModel.findOne(
+            {
+                username: data.username
+            },
+            function (error, user) {
+                var newUser;
+
+                if (error) {
+                    deferred.reject(error);
+                    return;
```
Comment (AndreySurzhan) [inline controllers/user.js]: are we going to stick to code style rules where between ```return``` should be an empty line?

File: fixtures/createUsers.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
+var mongoose = require('mongoose');
```
Comment (AndreySurzhan) [inline fixtures/createUsers.js]: sort

File: libs/log.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+            new winston.transports.Console({
+                colorize: true,
+                level: 'debug',
+                label: path
+            })
+        ]
+    });
+}
+
+module.exports = getLogger;
```
Comment (AndreySurzhan) [inline libs/log.js]: additinal line

File: libs/oauth2.js [hunk 4]
```diff
@@ -0,0 +1,273 @@
+var oauth2orize = require('oauth2orize');
```
Comment (AndreySurzhan) [inline libs/oauth2.js]: sort

File: models/auth/accessToken.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        required: true
+    },
+    token: {
+        type: String,
+        unique: true,
+        required: true
+    },
+    created: {
+        type: Date,
+        default: Date.now
```
Comment (AndreySurzhan) [inline models/auth/accessToken.js]: every time you save document based on this model you will refresh ```created``` date. Is it ok in that case?

File: controllers/user.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+var Q = require('q');
+
+var UserModel = require('../models/auth/user');
+
+var userController = {
+    /**
+     * Create new user
+     *
+     * @param {Object} data
+     * @param {Object} data.username - email in fact
```
Comment (AndreySurzhan) [inline controllers/user.js]: string

File: models/auth/user.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        type: String,
+        required: true
+    },
+    created: {
+        type: Date,
+        default: Date.now
+    }
+});
+
+UserSchema.methods.encryptPassword = function (password) {
```
Comment (AndreySurzhan) [inline models/auth/user.js]: js docs?

File: controllers/user.js (hunk 1 above)
Comment (wildeastengineer) [inline controllers/user.js]: yes, thanks

File: models/auth/accessToken.js (hunk 5 above)
Comment (wildeastengineer) [inline models/auth/accessToken.js]: I don't think so. Model get `Date.now` only the first time when it is created.

**Comment by wildeastengineer** [comment]
@AndreySurzhan thanks for notes. PR is updated

File: fixtures/createUsers.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+console.log('Remove all users');
+UserModel.remove({}, function () {
+    var i;
+    var user;
+
+    for (i = 0; i < users.length; i++) {
+        user = new UserModel(users[i]);
+        user.save(function (err, user) {
+            if (err) {
+                console.log('Error', err);
```
Comment (AndreySurzhan) [inline fixtures/createUsers.js]: Use logger

**Comment by AndreySurzhan** [comment]
👍

**Pull request merged**